### PR TITLE
issue #60 - get integration status from json returned by rest call

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

### CODE REVIEW

File: src/CamelKNodeProvider.ts [hunk 1]
```diff
@@ -196,7 +196,8 @@ export class CamelKNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 			let o = JSON.parse(temp);
 			for (var i=0; i<o.items.length;i++) {
 				var integrationName = o.items[i].metadata.name;
-				let newNode = new TreeNode("string", integrationName, 'Running', vscode.TreeItemCollapsibleState.None);
+				var status = o.items[i].status.phase;
```
Comment (apupier) [inline src/CamelKNodeProvider.ts]: what do you think to try to provide more explicit naming to `o` and `items[i]`?

I understood that after looking to the context but i'm not even sure.
`o` is the list of integrations represented as JSon?
`o.items[i]` is the Json for one integration?

Comment (apupier) [inline src/CamelKNodeProvider.ts]: I know it was already a problem before but when i read this code again, i really feel that it is so weak. We have no type-safety and neither automated test. If what is returned is modified we will not detect it.

Comment (apupier) [reply src/CamelKNodeProvider.ts]: created https://github.com/camel-tooling/vscode-camelk/issues/63 and https://github.com/camel-tooling/vscode-camelk/issues/62

Comment (bfitzpat) [reply src/CamelKNodeProvider.ts]: Improved

Comment (bfitzpat) [reply src/CamelKNodeProvider.ts]: We have no way to do integration testing at this time. Maybe we can mock it?

Comment (apupier) [reply src/CamelKNodeProvider.ts]: mocking will allow to test our code logic but not to detect that the returned values have changed.

**Pull request merged**